client/src/main/java/org/apache/rocketmq/client/impl/producer/DefaultMQProducerImpl.java

```diff
@@ -409,17 +410,32 @@ public MessageExt queryMessageByUniqKey(String topic, String uniqKey)
     /**
      * DEFAULT ASYNC -------------------------------------------------------
      */
-    public void send(Message msg,
-        SendCallback sendCallback) throws MQClientException, RemotingException, InterruptedException {
+    public void send(Message msg, SendCallback sendCallback) {
         send(msg, sendCallback, this.defaultMQProducer.getSendMsgTimeout());
     }
 
-    public void send(Message msg, SendCallback sendCallback, long timeout)
-        throws MQClientException, RemotingException, InterruptedException {
-        try {
-            this.sendDefaultImpl(msg, CommunicationMode.ASYNC, sendCallback, timeout);
-        } catch (MQBrokerException e) {
-            throw new MQClientException("unknownn exception", e);
+    public void send(final Message msg, final SendCallback sendCallback, final long timeout) {
+        this.getCallbackExecutor().submit(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    sendDefaultImpl(msg, CommunicationMode.ASYNC, sendCallback, timeout);
+                } catch (Exception e) {
+                    handleCallbackException(e, sendCallback);
```

Comment:
How to handle the `RejectedExecutionException`? While it has the similar effect as `NettyRemotingAbstract.semaphoreAsync` -:)

Reply:
please refer to https://github.com/spring-projects/spring-framework/blob/master/spring-context/src/main/java/org/springframework/scheduling/concurrent/ThreadPoolExecutorFactoryBean.java